### ISSUE TITLE
Serial stream issues with current Nim compiler

### ISSUE THREAD

**Issue by euantorano** [euantorano/serial.nim#20]
Because this module does'nt compile with my ubuntu nim version I just downloaded and compiled the current git version of nim and nimble two hours ago.

I get the following error: 

```
/home/bert/.nimble/pkgs/serial-1.1.2/serial/serialstream.nim(117, 17) Error: type mismatch: got <proc (s: Stream, buffer: pointer, bufLen: int): int{.gcsafe, locks: <unknown>.}> but expected 'proc (s: Stream, buffer: pointer, bufLen: int): int{.gcsafe.}'
```

Please help. I'm an absolute beginner and want to get into nim with learning by doing. But, sorry, this problem it to hard for me.

Alternatively I would be helpful if somebody can give me a nim-compiler version that runs with the current serial module.

Shouldn't the dependency section in the serial.nimble file be updated?

Thanks!!!!

_Originally posted by @WoodyAtHome in https://github.com/euantorano/serial.nim/issues/12#issuecomment-464872747_

**Comment by euantorano** [comment]
@WoodyAtHome it looks like this may be due to a change in the streams interface. I will look into this tomorrow after work and push a new version to resolve the issue.

**Comment by WoodyAtHome** [comment]
Great, thx

**Comment by euantorano** [comment]
Should be resolved by #21
